enterprise/server/api/api_server_test.go

```diff
@@ -443,7 +448,98 @@ func TestGetActionWithRealData(t *testing.T) {
 	require.Equal(t, "stderr", actionResp.GetAction()[0].GetFile()[0].GetName())
 }
 
-func getEnvAndCtx(t *testing.T, user string) (*testenv.TestEnv, context.Context) {
+func TestGetTargetWithRealData(t *testing.T) {
+	// Avoid trying to replicate the blobs to local test server
+	build_event_handler.PersistArtifactsEnabled = func() bool {
+		return false
+	}
```

Comment:
can you use flags.Set instead here? This is pretty icky messing with package level variables like this

Reply:
I went with flags.Set at first, but ran into a race detector issue so I tried using this approach (which did not resolve the issue). Reverted to flags.Set now.

I added some cleanup improvements to the event handler as well as the api_server_test code:
- Call handler.Stop and channel.Close after streamBuildFromTestData and streamBuild to avoid leaving resources behind between tests.

- Modified handler.Stop to also call statsRecorder.Stop and webhookNotifier.Stop to properly shutting down all the errgroups inside the handler.

I think these fixed the race in the test.